src/app/qgsdecorationitem.cpp

```diff
@@ -72,6 +75,7 @@ void QgsDecorationItem::saveToProject()
   QgsDebugMsg( "Entered" );
   QgsProject::instance()->writeEntry( mNameConfig, "/Enabled", mEnabled );
   QgsProject::instance()->writeEntry( mNameConfig, "/Placement", static_cast< int >( mPlacement ) );
+  QgsProject::instance()->writeEntry( mNameConfig, "/MarginUnits", static_cast< int >( mMarginUnits ) );
```

Comment:
As per line 69, this would become

    QgsProject::instance()->writeEntry( mNameConfig, "/MarginUnits", QgsSymbolLayerV2Utils::encodeOutputUnit( mMarginUnits ) );